### DISCUSSION TITLE
feature: Add Support for nested Object in form

### DISCUSSION THREAD

**Discussion by ahmedsakri** [manojadams/mui-forms#42] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, the schema only supports one level of `prop` support. i.e, if i pass `prop: "properties"` in the schema. The value will be saved inside the `properties` object.

Below is the current scenario:

Schema:
```
{
  "name": "name",
  "meta": {
    "displayName": "Name",
    "displayType": "text_field",
    "displayProps": {
      "md": 12,
      "sm": 12,
      "xs": 12
    },
    "validation": {
      "max": 60,
      "required": true
    }
  },
  "prop": "properties"
}
```

Submitted Form Data

```
properties: {
  name: "Sample Name"
}
```

**Describe the solution you'd like**
Required Output:
```
{
    "properties": {
        "name": "Sample Name",
        "retry_config": {
            "max_retries": 3,
            "retry_interval": 5
        }
    }
}
```

Json Schema:
```
{
  "name": "name",
  "meta": {
    "displayName": "Name",
    "displayType": "text_field",
    "displayProps": {
      "md": 12,
      "sm": 12,
      "xs": 12
    },
    "validation": {
      "max": 60,
      "required": true
    }
  },
  "prop": "properties"
},
{
    "name": "max_retries",
    "meta": {
        "displayName": "Max Retries (m/s)",
        "displayType": "number",
        "displayProps": {
            "md": 12,
            "sm": 12,
            "xs": 12
        },
        "validation": {
            "max": 60,
            "required": true
        }
    },
    "prop": "properties.retry_config"
},
{
    "name": "retry_interval",
    "meta": {
        "displayName": "Retry Interval (m/s)",
        "displayType": "number",
        "displayProps": {
            "md": 12,
            "sm": 12,
            "xs": 12
        },
        "validation": {
            "max": 60,
            "required": true
        }
    },
    "prop": "properties.retry_config"
}
```

I am not able to pass something like this: `"prop": "properties.retry_config"`

**Describe alternatives you've considered**
Tried using `"prop": "properties.retry_config"`, but it did not worked.

**Additional context**
N/A

**Comment by ahmedsakri** [comment]
@manojadams Can you please check if we can add this?

**Comment by manojadams** [comment]
Try this format: prop#nested_prop#nested_prop

**Comment by manojadams** [comment]
Hi @ahmedsakri refer to the example here:
https://mui-forms.vercel.app/fields/general-properties#3-example-with-nested-prop

Closing it as the requested feature is already implemented.

**Comment by ahmedsakri** [comment]
@manojadams Yes this works perfectly. I was wondering why `#` was selected instead of just `.` for nested values? It would be easier for dev too to understand.

Nevertheless, Your solution works!
